tests/integration.rs

```diff
@@ -341,3 +372,89 @@ fn subcommand_usage() -> TestResult {
 fn standalone_usage() -> TestResult {
     golden_reference(&["-h"], "tests/standalone-usage.txt")
 }
+
+/// This test is a bit more verbose than the other tests because it uses a slightly
+/// different mechanism for setting up and sweeping / cleaning.
+///
+/// Step-by-step:
+///
+/// * Create a new temporary directory. Other tests only output the built files into
+///   the target directory, but this test copies the whole template project into the
+///   target directory.
+/// * Build the two different (nested) projects using cargo build, but without incremental
+///   compilation, because cargo-sweep doesn't remove files built by incremental compilation,
+///   which affects this test since this test doesn't use a separate target folder for just
+///   the target files.
+/// * Do a dry run sweep, asserting that nothing was actually cleaned.
+/// * Do a proper sweep, asserting that the target directory size is back down to its
+///   expected size.
+///
+/// Please note that there is some ceremony involved, namely correctly setting:
+/// * `CARGO_TARGET_DIR` to `unset`, so as not to clash with users who use `CARGO_TARGET_DIR`
+///   when they invoke `cargo test` for running these tests.
+/// * `CARGO_INCREMENTAL` to `0`, because cargo-sweep doesn't yet clear incremental files.
+#[test]
+fn recursive_multiple_root_workspaces() -> TestResult {
+    let target = tempdir()?;
```

Comment:
This is ambiguous with the `target/` directory cargo generates - please call this `temp_workspace_dir` or something like that.

Reply:
Done